src/map.cpp

```diff
@@ -1378,6 +1378,25 @@ s16 ServerMap::getWaterLevel()
 	return getMapgenParams()->water_level;
 }
 
+bool ServerMap::saoPosOverMapgenLimit(const v3f &p)
+{
+	// -3100/31000 for testing, instead need to get
+	// 'terrain_edge_min/max' from 'class Mapgen'
+	float sao_limit_min = -31000 * BS - 3.0f;
```

Comment:
This is code for testing (see comment), will be changed. The min and max world edges will be different.